File: Task 2/Java/app/src/main/java/dev/ogabek/java/activity/MainActivity.java
package dev.ogabek.java.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.gson.Gson;

import dev.ogabek.java.R;
import dev.ogabek.java.manager.PrefsManager;
import dev.ogabek.java.model.User;

public class MainActivity extends AppCompatActivity {

    private EditText fullName, email, password, confirmPassword;
    private TextView data;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

    }

    private void initViews() {

        fullName = findViewById(R.id.et_fullName);
        email = findViewById(R.id.et_email);
        password = findViewById(R.id.et_password);
        confirmPassword = findViewById(R.id.et_confirmPassword);

        Button save = findViewById(R.id.btn_save);
        Button load = findViewById(R.id.btn_load);

        data = findViewById(R.id.tv_data);

        save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                User user = new User(fullName.getText().toString(), email.getText().toString(), password.getText().toString(), confirmPassword.getText().toString());

                new PrefsManager(MainActivity.this).setData("user", new Gson().toJson(user));

                Toast.makeText(MainActivity.this, "Saved", Toast.LENGTH_SHORT).show();
                fullName.setText("");
                email.setText("");
                password.setText("");
                confirmPassword.setText("");
            }
        });

        load.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                User user = new Gson().fromJson(new PrefsManager(MainActivity.this).getStringData("user"), User.class);

                if (user != null) {
                    String account =
                            "Full Name : " + user.getFullName() +
                            "\nEmail : " + user.getEmail() +
                            "\nPassword : " + user.getPassword() +
                            "\nConfirm Password : " + user.getConfirmPassword();

                    data.setText(account);
                }
            }
        });

    }
}

File: Task 2/Kotlin/app/src/main/java/dev/ogabek/kotlin/activity/MainActivity.kt
package dev.ogabek.kotlin.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.google.gson.Gson
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.User
import dev.ogabek.kotlin.manager.PrefsManager

class MainActivity : AppCompatActivity() {
    private lateinit var fullName: EditText
    private lateinit var email: EditText
    private lateinit var password: EditText
    private lateinit var confirmPassword: EditText
    private lateinit var data: TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViews()
    }

    private fun initViews() {
        fullName = findViewById(R.id.et_fullName)
        email = findViewById(R.id.et_email)
        password = findViewById(R.id.et_password)
        confirmPassword = findViewById(R.id.et_confirmPassword)
        val save = findViewById<Button>(R.id.btn_save)
        val load = findViewById<Button>(R.id.btn_load)
        data = findViewById(R.id.tv_data)
        save.setOnClickListener {

            val user = User(fullName.text.toString(), email.text.toString(), password.text.toString(), confirmPassword.text.toString())

            PrefsManager(this@MainActivity).setData("user", Gson().toJson(user))
            Toast.makeText(this@MainActivity, "Saved", Toast.LENGTH_SHORT).show()
            fullName.setText("")
            email.setText("")
            password.setText("")
            confirmPassword.setText("")
        }
        load.setOnClickListener {
            val user = Gson().fromJson(PrefsManager(this@MainActivity).getStringData("user"), User::class.java)

            if (user != null) {
                val account = """
                Full Name : ${user.fullName}
                Email : ${user.email}
                Password : ${user.password}
                Confirm Password : ${user.confirmPassword}
                """.trimIndent()

                data.text = account
            }
        }
    }
}